app/src/main/java/io/appium/uiautomator2/model/UiObject2Element.java

```diff
@@ -98,77 +100,107 @@ public String getText() {
-        throw generateNoAttributeException(attr);
-    }
 
-    public boolean getBoolAttribute(final String attr) throws UiAutomator2Exception {
-        switch (attr) {
+        final Object result;
+        switch (attr.toLowerCase()) {
+            case "name":
+            case "text":
+                result = getText();
+                break;
+            case "contentdescription":
+                result = element.getContentDescription();
+                break;
+            case "classname":
+                result = element.getClassName();
+                break;
+            case "resourceid":
+            case "resource-id":
+                result = element.getResourceName();
+                break;
+            case "contentsize":
+                result = ElementHelpers.getContentSize(this);
+                break;
             case "enabled":
-                return element.isEnabled();
+                result = element.isEnabled();
+                break;
             case "checkable":
-                return element.isCheckable();
+                result = element.isCheckable();
+                break;
             case "checked":
-                return element.isChecked();
+                result = element.isChecked();
+                break;
             case "clickable":
-                return element.isClickable();
+                result = element.isClickable();
+                break;
             case "focusable":
-                return element.isFocusable();
+                result = element.isFocusable();
+                break;
             case "focused":
-                return element.isFocused();
-            case "longClickable":
-                return element.isLongClickable();
+                result = element.isFocused();
+                break;
+            case "longclickable":
+                result = element.isLongClickable();
+                break;
             case "scrollable":
-                return element.isScrollable();
+                result = element.isScrollable();
+                break;
             case "selected":
-                return element.isSelected();
+                result = element.isSelected();
+                break;
             case "displayed":
-                return AccessibilityNodeInfoGetter.fromUiObject(element) != null;
+                result = AccessibilityNodeInfoGetter.fromUiObject(element) != null;
+                break;
             case "password":
                 AccessibilityNodeInfo nodeInfo = AccessibilityNodeInfoGetter.fromUiObject(element);
-                return nodeInfo != null && nodeInfo.isPassword();
+                result = nodeInfo != null && nodeInfo.isPassword();
+                break;
             default:
                 throw generateNoAttributeException(attr);
         }
+        if (result == null) {
+            return null;
+        }
+        return (result instanceof String) ? (String) result : String.valueOf(result);
```

Comment:
if `result` is already a string, why do we cast it to one?

Reply:
how can a boolean be an instance of a String?

Reply:
it cannot. That is why we have Sttring.valueOf

Reply:
the part that i'm talking about is different. the conditional checks whether `result instanceof String`. if that is true, we return `(String) result`. I'm saying that we shouldn't need to cast `result` to a `String` in the case where we've already confirmed that it is a string.

Reply:
got it, thanks. makes you want stuff like Swift where once you do a check you can use the assumptions guaranteed by the check inside the block.